### DISCUSSION TITLE
licensing

### DISCUSSION THREAD

**Discussion by rxjx** [autogluon/autogluon#2228] [Q&A] [answered]
Forgive my ignorance around these issues but does Autogluon's apache license also extend to the embeddings it uses? Will I get into licensing trouble if I use any of the Hugging Face embeddings used in autogluon via autogluon function calls?

**Comment by Innixma** [comment, marked as answer]
HuggingFace is also Apache 2 licensed, meaning you should not run into issues.

**Reply by rxjx** [reply]
That's true but the licensing for individual embeddings e.g. CLIP is not clear.

**Reply by rxjx** [reply]
That's true but the licensing for individual embeddings e.g. CLIP is not clear.

**Reply by Innixma** [reply]
That would be a good question to ask the HuggingFace maintainers, I would be interested to know their answer.

**Reply by rxjx** [reply]
It's unclear and there are no recent responses:
https://discuss.huggingface.co/t/license-for-models-on-huggingface/13373/3 
https://discuss.huggingface.co/t/license-of-flair-ner-english-ontonotes-fast/8767
https://discuss.huggingface.co/t/facebook-opt-30b-license/17979

**Reply by Innixma** [reply]
Unfortunately I won't be able to provide an authoritative answer on this topic, and you should seek answers from HuggingFace in this regard. However, my understanding is that we only use Apache/MIT/BSD (permissive) licensed models from HuggingFace by default. Because AutoGluon supports any HuggingFace compatible model, you can also specify to use one that you trust is licensed to your requirements within AutoGluon to avoid any issues.

@sxjscience If you know more regarding licensing of HF models please feel free to respond

**Reply by rxjx** [reply]
Thank you for looking into this. I appreciate your help.
